compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -505,45 +549,155 @@ struct RstSignatureVisitor {
     interpose(xs.begin(), xs.end(), separator, surroundBegin, surroundEnd);
   }
 
-  bool enter(const Identifier* r) {
-    os_ << r->name().c_str();
+  template<typename T>
+  bool enterLiteral(const T* l) {
+    os_ << l->text().c_str();
     return false;
   }
 
-  bool enter(const Record* r) {
-    // TODO: Shouldn't this be record, not Record?
-    if (textOnly_) os_ << "Record: ";
-    os_ << r->name().c_str();
-    return false;
+    bool isPostfix(const OpCall* op) {
+    return (op->isUnaryOp() &&
+            (op->op() == USTR("postfix!") || op->op() == USTR("?")));
   }
 
-  bool enter(const Class* c) {
-    os_ << c->name().c_str();
-    if (c->parentClass()) {
-      os_ << " : ";
-      c->parentClass()->traverse(*this);
+  /*
+    helper for printing binary op calls, special handling for keyword operators
+    to conditionally add spaces around the operator
+  */
+  void printBinaryOp(const OpCall* node) {
+    assert(node->numActuals() == 2);
+    bool addSpace = wantSpaces(node->op(), printingType_) ||
+                    node->op() == USTR("by") ||
+                    node->op() == USTR("align") ||
+                    node->op() == USTR("reduce=") ||
+                    node->op() == USTR("reduce") ||
+                    node->op() == USTR("scan") ||
+                    node->op() == USTR("dmapped");
+    opHelper(node, 0, false);
+    if (addSpace && node->op() != USTR(":"))
+      os_ << " ";
+    os_ << node->op();
+    if (addSpace)
+      os_ << " ";
+    opHelper(node, 1, false);
+  }
+
+  /*
+    helper for printing unary op calls, special handling for nilable operator
+    to conditionally prepend nilable in replacement of the ? operator when the
+    actual is not a function call
```

Comment:
As we discussed, we don't actually want to do that. But, I'm expecting the follow-up PR to fix it.

Reply:
yep, dropped it in the next PR